File: tools/generate_data_manifests.py
#!/usr/bin/env python3
"""
Simple manifest generator for Rising Flow data folder
Usage: python tools/generate_data_manifests.py
It will scan data/events folder and create data/_manifests/events.json
"""

import os
import json
from pathlib import Path

def list_folders(directory):
    """List all subdirectories in a given directory."""
    try:
        if not os.path.exists(directory):
            return []
        return [f for f in os.listdir(directory) 
                if os.path.isdir(os.path.join(directory, f)) and not f.startswith('_')]
    except Exception:
        return []

def ensure_dir(directory):
    """Ensure a directory exists, create if it doesn't."""
    os.makedirs(directory, exist_ok=True)

def main():
    # Get the script's directory and navigate to project root
    script_dir = Path(__file__).parent
    project_root = script_dir.parent
    
    data_root = project_root / 'data'
    manifests_dir = data_root / '_manifests'
    
    # Ensure manifests directory exists
    ensure_dir(manifests_dir)
    
    # Scan events directory
    events_dir = data_root / 'events'
    
    # Get all event folders (excluding template and manifest folders)
    all_events = list_folders(events_dir)
    
    # Create manifest
    manifest = {
        'upcoming': all_events,
        'past': []
    }
    
    # Write manifest file
    manifest_path = manifests_dir / 'events.json'
    with open(manifest_path, 'w', encoding='utf-8') as f:
        json.dump(manifest, f, indent=2, ensure_ascii=False)
    
    print(f'Wrote data/_manifests/events.json')
    print(f'Found {len(all_events)} event(s)')

if __name__ == '__main__':
    main()

File: tools/generate_data_manifests.js
#!/usr/bin/env node
// Simple manifest generator for Rising Flow data folder
// Usage: node tools/generate_data_manifests.js
// It will scan data/events/upcoming and data/events/past folders and create data/_manifests/events.json

const fs = require('fs');
const path = require('path');

const dataRoot = path.join(__dirname, '..', 'data');
const manifestsDir = path.join(dataRoot, '_manifests');

function listFolders(p) {
  try {
    return fs.readdirSync(p).filter(f => fs.statSync(path.join(p, f)).isDirectory());
  } catch (e) {
    return [];
  }
}

function ensureDir(p) {
  if (!fs.existsSync(p)) fs.mkdirSync(p, { recursive: true });
}

function main() {
  ensureDir(manifestsDir);

  const upcomingDir = path.join(dataRoot, 'events', 'upcoming');
  const pastDir = path.join(dataRoot, 'events', 'past');

  const upcoming = listFolders(upcomingDir);
  const past = listFolders(pastDir);

  const manifest = { upcoming, past };
  fs.writeFileSync(path.join(manifestsDir, 'events.json'), JSON.stringify(manifest, null, 2));
  console.log('Wrote data/_manifests/events.json');
}

main();